features/steps/ccx_messaging.py

```diff
@@ -0,0 +1,309 @@
+# Copyright © 2024 Jiri Papousek, Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of common steps for all services using ccx-messaging module."""
+
+import gzip
+import os
+import subprocess
+
+import yaml
+from behave import given, then, when
+from kafka.cluster import ClusterMetadata
+from src import kafka_util
+
+SERVICES = {"SHA extractor": "insights_sha_extractor", "DVO extractor": "dvo_extractor"}
+
+
+def transform_service_name(service_name):
+    """Transform service name used in test steps to name used in dictionaries and config names."""
+    return SERVICES[service_name]
+
+
+@given("{service} service is not started")
+def service_not_started(context, service):
+    """Check if SHA extractor service has been started."""
+    service_name = transform_service_name(service)
+    assert (
+        not hasattr(context, "services") or service_name not in context.services.keys()
+    )
+
+
+@given(
+    'Kafka broker is started on host and port specified '
+    'in {service} configuration "{compression_var}"',
```

Comment:
I'm lost about what this compression_var is and why the configuration file would be compressed

Reply:
I think it's used to start the service with/without compression like in
```
    Given Kafka broker is started on host and port specified in SHA extractor configuration "no-compression"
```

Reply:
This one was part of SHA extractor tests done by Jakub. But yes, Juan is right, there are two configurations, one with enabled message compressions and one without.

Reply:
It's pretty error prone (an entire config file is loaded just to enable compression), but we can solve that in a different PR,